### PR TITLE
Fixes for SU group

### PR DESCRIPTION
**Pull request by imbirik**
fixes related to incorrect handling of complex lie groups.

### CODE REVIEW

File: geometric_kernels/feature_maps/random_phase.py [hunk 1]
```diff
@@ -74,20 +74,26 @@ def __call__(
...
+        random_phases_b = B.cast(dtype, from_numpy(X, random_phases))
 
         phi_product = self.kernel.eigenfunctions.phi_product(
             X, random_phases_b, **params
         )  # [N, O, L]
 
-        embedding = B.cast(B.dtype(params["lengthscale"]), phi_product)  # [N, O, L]
-        weights_t = B.cast(B.dtype(params["lengthscale"]), B.transpose(weights))
+        embedding = B.cast(dtype, phi_product)  # [N, O, L]
+        weights_t = B.cast(dtype, B.transpose(weights))
```
Comment (vabor112) [inline geometric_kernels/feature_maps/random_phase.py:91]: Do we need to cast weights to be complex?

**Review by vabor112 — COMMENTED** [review]
I tried running `notebooks/SpecialUnitary.ipynb` but the visualized kernels still does not make sense. Something's still off.

File: geometric_kernels/feature_maps/random_phase.py (hunk 1 above)
Comment (imbirik) [reply geometric_kernels/feature_maps/random_phase.py:91]: We need to cast them into the same type as everything else, so B.cast(dtype,...) is the most intuitive way to do this. But, yes, it is enough to cast them into the corresponding float type.

**Review by vabor112 — APPROVED** [review]
LGTM now

**Pull request merged**